src/app/stream-deck/actions.ts

```diff
@@ -0,0 +1,366 @@
+import { createLoadoutShare } from 'app/dim-api/dim-api';
+import { startFarming, stopFarming } from 'app/farming/actions';
+import { t } from 'app/i18next-t';
+import { DimItem } from 'app/inventory/item-types';
+import { moveItemTo } from 'app/inventory/move-item';
+import {
+  allItemsSelector,
+  currentStoreSelector,
+  storesSelector,
+  vaultSelector,
+} from 'app/inventory/selectors';
+import { DimStore } from 'app/inventory/store-types';
+import { hideItemPopup } from 'app/item-popup/item-popup';
+import { maxLightLoadout, randomLoadout } from 'app/loadout-drawer/auto-loadouts';
+import { applyLoadout } from 'app/loadout-drawer/loadout-apply';
+import { convertDimLoadoutToApiLoadout } from 'app/loadout-drawer/loadout-type-converters';
+import { Loadout, LoadoutItem } from 'app/loadout-drawer/loadout-types';
+import { pullFromPostmaster } from 'app/loadout-drawer/postmaster';
+import { loadoutsSelector } from 'app/loadout-drawer/selectors';
+import { d2ManifestSelector } from 'app/manifest/selectors';
+import { showNotification } from 'app/notifications/notifications';
+import { setSearchQuery } from 'app/shell/actions';
+import { refresh } from 'app/shell/refresh-events';
+import { RootState, ThunkResult } from 'app/store/types';
+import { checkAuthorization } from 'app/stream-deck/authorization/authorization';
+import encryption from 'app/stream-deck/authorization/encryption';
+import { streamDeckLocal } from 'app/stream-deck/util/local-storage';
+import packager from 'app/stream-deck/util/packager';
+import { observeStore } from 'app/utils/redux-utils';
+import { BucketHashes } from 'data/d2/generated-enums';
+import _ from 'lodash';
+import { createAction } from 'typesafe-actions';
+
+export let streamDeckWebSocket: WebSocket;
+
+let refreshInterval: ReturnType<typeof setInterval>;
+
+// Cache shares to loadouts weakly, to cut down on creating shares
+const loadoutShares = new WeakMap<Loadout, string>();
+
+export const streamDeckConnected = createAction('stream-deck/CONNECTED')();
+
+export const streamDeckDisconnected = createAction('stream-deck/DISCONNECTED')();
+
+export const streamDeckChangeStatus = createAction('stream-deck/CHANGE-STATUS')<boolean>();
+
+export const streamDeckWaitSelection = createAction('stream-deck/WAIT-SELECTION')<
+  'loadout' | 'item'
+>();
+
+export const streamDeckClearSelection = createAction('stream-deck/CLEAR-SELECTION')();
+
+// serialize the data and send it if connected
+export function sendToStreamDeck(args: Record<string, any>): ThunkResult {
+  return async () => {
+    if (streamDeckWebSocket?.readyState === WebSocket.OPEN) {
+      const sharedKey = streamDeckLocal.sharedKey();
+      sharedKey && streamDeckWebSocket.send(encryption.encrypt(JSON.stringify(args), sharedKey));
+    }
+  };
+}
+
+// on click on InventoryItem send the selected item to the Stream Deck
+export function streamDeckSelectItem(item: DimItem): ThunkResult {
```

Comment:
Gotcha, so this is to enable selecting the item in order to set the target of an action on a streamdeck button.